src/utils/documentdb_model.ts

```diff
@@ -47,51 +83,53 @@ export abstract class DocumentDbModel<
    *     resolves to either an error or the retrieved version of the persisted
    *     document.
    */
-  public async create(
-    document: TN,
-    partitionKey: string
-  ): Promise<Either<DocumentDb.QueryError, TR>> {
-    // we don't want to store the "kind" property, let's remove it
-    const kindlessDocument = Object.assign(Object.assign({}, document), {
-      kind: undefined
-    });
-
-    // attempt to persist the document
-    const maybeCreatedDocument = await DocumentDbUtils.createDocument(
-      this.dbClient,
-      this.collectionUri,
-      kindlessDocument,
-      partitionKey
-    );
-
-    // if the result is successful we map it to a TR type
-    return maybeCreatedDocument.map(this.toRetrieved);
+  public create(
+    document: T,
+    options?: RequestOptions
+  ): TaskEither<QErrors, t.TypeOf<TR<T>>> {
+    const createItem = this.container.items.create;
+    const item = this.type.encode(document);
+    return tryCatch<QErrors, PromiseType<ReturnType<typeof createItem>>>(
+      () => createItem(item, options),
+      e => ({
+        error: toError(e),
+        kind: "RESPONSE"
+      })
+    )
+      .map(_ => _.resource)
+      .filterOrElse(isDefined, { kind: "EMPTY" }) // TODO: not sure whether an undef resource should be an error
```

Comment:
Maybe for a new document, we should return an error, because it' s similar to insert a tuple with all columns set to null, in the relational world isn't it?

Reply:
that's what I don't know yet, I suspect `undefined` is part of the response type used for all methods but not actually returned by `create`, have to check the sdk code to confirm